src/Pool.sol

```diff
@@ -267,6 +267,40 @@ abstract contract Pool {
         emit Collected(receiverAddr, collected, dripped);
     }
 
+    /// @notice Counts cycles which will need to be analyzed when collecting or flushing.
+    /// This function can be used to detect that there are too many cycles
+    /// to analyze in a single transaction and flushing is needed.
+    /// @param receiverAddr The address of the receiver
+    /// @return flushable The number of cycles which can be flushed
+    function flushableCycles(address receiverAddr) public view returns (uint64 flushable) {
+        uint64 nextCollectedCycle = receivers[receiverAddr].nextCollectedCycle;
+        if (nextCollectedCycle == 0) return 0;
+        uint64 currFinishedCycle = _currTimestamp() / cycleSecs;
+        return currFinishedCycle + 1 - nextCollectedCycle;
+    }
+
+    /// @notice Flushes uncollected cycles of the user.
+    /// Flushed cycles won't need to be analyzed when the user collects from them.
+    /// Calling this function does not collect and does not affect the collectable amount.
+    ///
+    /// This function is needed when collecting funds received over a period so long, that the gas
+    /// needed for analyzing all the uncollected cycles can't fit in a single transaction.
+    /// Calling this function allows spreading the analysis cost over multiple transactions.
+    /// A cycle is never flushed more than once, even if this function is called many times.
+    /// @param receiverAddr The address of the receiver
+    /// @param maxCycles The maximum number of flushed cycles.
+    /// If too low, flushing will be cheap, but will cut little gas from the next collection.
+    /// If too high, flushing may become too expensive to fit in a single transaction.
+    /// @return flushable The number of cycles which can be flushed
+    function flushCycles(address receiverAddr, uint64 maxCycles) public returns (uint64 flushable) {
+        Receiver storage receiver = receivers[receiverAddr];
+        flushable = flushableCycles(receiverAddr);
+        uint64 flushed = maxCycles < flushable ? maxCycles : flushable;
```

Comment:
```suggestion
        uint64 cycles = maxCycles < flushable ? maxCycles : flushable;
```
The past tense is a bit confusing because the `flush` didn't happen yet.

Reply:
Fixed